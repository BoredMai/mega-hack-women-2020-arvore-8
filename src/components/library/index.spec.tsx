import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { useHistory, useRouteMatch } from 'react-router-dom';

import Library from '.';

jest.mock('react-router-dom', () => ({
  useHistory: jest.fn().mockName('useHistory'),
  useRouteMatch: jest.fn().mockName('useRouteMatch'),
}));
jest.useFakeTimers();

describe('<Library>', () => {
  const book = { icon: 'princesa', label: 'Princesa', url: 'princesa' };
  const history = { push: jest.fn() };
  const path = 'path';
  let wrapper: ShallowWrapper;

  (useHistory as jest.Mock).mockReturnValue(history);
  (useRouteMatch as jest.Mock).mockReturnValue({ path });

  it('renders loading state', () => {
    wrapper = shallow(<Library />) as ShallowWrapper;
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  describe('after fetching data', () => {
    beforeEach(() => {
      wrapper = shallow(<Library />) as ShallowWrapper;
      jest.runAllTimers();
    });

    it('renders properly', () => {
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('selects book on item click', () => {
      wrapper.find('BookItem').at(0).simulate('selectBook', book);

      expect(wrapper.find('button')).toHaveLength(1);
    });

    it('navigates to selected book on button click', () => {
      wrapper.find('BookItem').at(0).simulate('selectBook', book);
      wrapper.find('button').simulate('click');

      expect(history.push).toHaveBeenCalledWith(`${path}/${book.url}`);
    });
  });
});
