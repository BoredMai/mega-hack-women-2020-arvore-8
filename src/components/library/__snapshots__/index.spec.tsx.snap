// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Library> after fetching data renders properly 1`] = `
<div
  className="grass"
>
  <ContentWrapper>
    <h1>
      Colha uma História
    </h1>
    <p>
      Muito bem, Clarice: Você está a um passo de ser uma grande autora!
    </p>
    <p>
      Agora, vamos escolher o personagem para a sua história?
    </p>
    <Shelf
      items={
        Array [
          Object {
            "icon": "princess.png",
            "label": "Princesa",
            "url": "princesa",
          },
          Object {
            "disabled": true,
            "icon": "wizard.png",
            "label": "Mago",
            "url": "mago",
          },
          Object {
            "disabled": true,
            "icon": "unicorn.png",
            "label": "Unicórnio",
            "url": "unicornio",
          },
        ]
      }
      onConfirm={[Function]}
    />
  </ContentWrapper>
</div>
`;

exports[`<Library> renders loading state 1`] = `
<div
  className="grass"
>
  <LoadingSpinner />
</div>
`;
